federation/client/clientset_generated/federation_internalclientset/typed/core/unversioned/fake/fake_core_client.go

```diff
@@ -46,8 +47,8 @@ func (c *FakeCore) Services(namespace string) unversioned.ServiceInterface {
 	return &FakeServices{c, namespace}
 }
 
-// GetRESTClient returns a RESTClient that is used to communicate
+// RESTClient returns a RESTClient that is used to communicate
 // with API server by this client implementation.
-func (c *FakeCore) GetRESTClient() *restclient.RESTClient {
-	return nil
+func (c *FakeCore) RESTClient() restclient.Interface {
+	return &fake.RESTClient{}
```

Comment:
We can't. ``pkg/controller/deployment/deployment_controller_test.go`` calls ``NewDeploymentController`` with fake rest client. Which calls ``client.Core().RESTClient().GetRateLimiter()``. If the ``RESTClient()`` returns ``nil``, ``client.Core().RESTClient().GetRateLimiter()`` fails with nil pointer panic. That was the reason why I return the fake RESTClient (thought not initialized).